app/src/main/java/com/fstyle/androidtrainning/widget/AutoScrollViewPager.java

```diff
@@ -0,0 +1,355 @@
+package com.fstyle.androidtrainning.widget;
+
+import android.content.Context;
+import android.os.Handler;
+import android.os.Message;
+import android.support.v4.view.MotionEventCompat;
+import android.support.v4.view.PagerAdapter;
+import android.support.v4.view.ViewPager;
+import android.util.AttributeSet;
+import android.view.MotionEvent;
+import android.view.animation.Interpolator;
+import java.lang.ref.WeakReference;
+import java.lang.reflect.Field;
+
+/**
+ * Created by Admin on 12/26/17.
+ */
+
+public class AutoScrollViewPager extends ViewPager {
+
+    public static final int DEFAULT_INTERVAL = 1500;
+
+    public static final int LEFT = 0;
+    public static final int RIGHT = 1;
+
+    /** do nothing when sliding at the last or first item **/
+    public static final int SLIDE_BORDER_MODE_NONE = 0;
+    /** cycle when sliding at the last or first item **/
+    public static final int SLIDE_BORDER_MODE_CYCLE = 1;
+    /** deliver event to parent when sliding at the last or first item **/
+    public static final int SLIDE_BORDER_MODE_TO_PARENT = 2;
+
+    /** auto scroll time in milliseconds, default is {@link #DEFAULT_INTERVAL} **/
+    private long interval = DEFAULT_INTERVAL;
+    /** auto scroll direction, default is {@link #RIGHT} **/
+    private int direction = RIGHT;
+    /**
+     * whether automatic cycle when auto scroll reaching the last or first item, default is true
+     **/
+    private boolean isCycle = true;
+    /** whether stop auto scroll when touching, default is true **/
+    private boolean stopScrollWhenTouch = true;
+    /**
+     * how to process when sliding at the last or first item, default is {@link
+     * #SLIDE_BORDER_MODE_NONE}
+     **/
+    private int slideBorderMode = SLIDE_BORDER_MODE_NONE;
+    /** whether animating when auto scroll at the last or first item **/
+    private boolean isBorderAnimation = true;
```

Comment:
WARNING: [AvoidFieldNameMatchingMethodName] Field isBorderAnimation has the same name as a method
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#AvoidFieldNameMatchingMethodName